### PR TITLE
docs/integration: More verbose and helpful description of u-boot support

### PR DESCRIPTION
**Pull request by ejoerns**
Latest questions on both the RAUC GitHub issue tracker and the mailing
list have shown that the current U-Boot integration description is too
unverbose and lacks some important hints.

This patch should improve the situation by giving more hints on what is
required and what is expected for letting RAUC interact with U-Boot.

Signed-off-by: Enrico Joerns <ejo@pengutronix.de>

### CODE REVIEW

File: docs/integration.rst [hunk 1]
```diff
@@ -488,29 +488,83 @@ successfully set up bootchooser to boot your slot::
...
 ~~~~~~
 
+To enable handling of redundant booting in U-Boot, manual scripting is
+required.
+U-Boot allows storing and modifying variables in its *Envionment*.
+Properly configured it can be accessed both from U-Boot itself as
+well as from Linux userspace.
+
+The RAUC U-Boot boot selection implementation uses a custom U-Boot script
+together withe the environment for managing and persisting slot selection.
```
Comment (mnhu) [inline docs/integration.rst]: `s/withe/with`

File: docs/integration.rst [hunk 2]
```diff
@@ -488,29 +488,83 @@ successfully set up bootchooser to boot your slot::
...
-To enable handling of redundant booting in U-Boot, manual scripting is required.
+Set up U-Boot Environment for RAUC
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The U-Boot bootloader interface of RAUC will rely on setting the U-Boot
-environment variables ``BOOT_<bootname>_LEFT`` which should mark the number of
-remaining boot attempts for the respective slot in your bootloader script.
+environment variables:
+
+* ``BOOT_ORDER``, which will contain the boot targets in the order they should
```
Comment (mnhu) [inline docs/integration.rst]: `s/the boot targets/a space separated list of boot targets`

File: docs/integration.rst [hunk 3]
```diff
@@ -488,29 +488,83 @@ successfully set up bootchooser to boot your slot::
...
+  be tried.
+* ``BOOT_<bootname>_LEFT``, which contains the number of remaining boot
+  attempts to perform for the respective slot
 
-To enable reading and writing of the U-Boot environment, you need to have the
-U-Boot target tool ``fw_setenv`` available on your devices rootfs.
+An U-Boot script example for handling redundant A/B boot setups is located in
```
Comment (mnhu) [inline docs/integration.rst]: `s/An U-Boot script example/An example U-Boot script`

File: docs/integration.rst [hunk 4]
```diff
@@ -488,29 +488,83 @@ successfully set up bootchooser to boot your slot::
...
+You must integrate your boot selection script into U-Boot.
+Refer the
+`U-Boot Scripting Capabilities <https://www.denx.de/wiki/DULG/UBootScripts>`_
+chapter in the U-Boot user documentation on how to achieve this.
+
+The script uses the names ``A`` and ``B`` as the ``bootname`` for the two
+different boot targets.
+Thus the resulting boot attempts variables will be ``BOOT_A_LEFT`` and
+``BOOT_B_LEFT``.
+The ``BOOT_ODER`` variable will contain ``A B`` if ``A`` is the primary slot or
```
Comment (mnhu) [inline docs/integration.rst]: `s/BOOT_ODER/BOOT_ORDER`

**Comment by ejoerns** [comment]
Thanks for the review! I've pushed an update with the changes you mentioned.

**Pull request merged**